packages/react-vapor/src/components/pagination/PaginationPagesNumber.tsx

```diff
@@ -0,0 +1,156 @@
+import {IReactVaporState, IReduxActionsPayload} from '../../ReactVapor';
+import {IReduxAction} from '../../utils/ReduxUtils';
+import {
+    INavigationPaginationDispatchProps,
+    INavigationPaginationOwnProps,
+    INavigationPaginationStateProps,
+} from '../navigation/pagination/NavigationPagination';
+import {addPagination, changePage, removePagination} from '../navigation/pagination/NavigationPaginationActions';
+import {IPaginationState} from '../navigation/pagination/NavigationPaginationReducers';
+import {NavigationPaginationSelect} from '../navigation/pagination/NavigationPaginationSelect';
+import {Svg} from '../svg/Svg';
+
+export interface IPaginationPagesNumberOwnProps {
+    id?: string;
+    totalPages: number;
+    numberOfPagesToShow?: number;
+    previousLabel?: string;
+    nextLabel?: string;
+    loadingIds?: string[];
+    hidden?: boolean;
+    disabled?: boolean;
+}
+
+const mapStateToProps = (
+    state: IReactVaporState,
+    ownProps: INavigationPaginationOwnProps
+): INavigationPaginationStateProps => {
+    const item: IPaginationState = _.findWhere(state.paginationComposite, {id: ownProps.id});
+
+    return {
+        currentPage: item ? item.pageNb : 0,
+    };
+};
+
+const mapDispatchToProps = (
+    dispatch: (action: IReduxAction<IReduxActionsPayload>) => void,
+    ownProps: INavigationPaginationOwnProps
+): INavigationPaginationDispatchProps => ({
+    onRender: () => dispatch(addPagination(ownProps.id)),
+    onDestroy: () => dispatch(removePagination(ownProps.id)),
+    onPageClick: (pageNb: number) => dispatch(changePage(ownProps.id, pageNb)),
+});
+
+export interface INavigationPaginationProps
+    extends IPaginationPagesNumberOwnProps,
+        ReturnType<typeof mapDispatchToProps>,
+        ReturnType<typeof mapStateToProps> {}
+
+export const NUMBER_OF_PAGES_SHOWING: number = 7;
+export const PREVIOUS_LABEL: string = 'Previous';
+export const NEXT_LABEL: string = 'Next';
+
+class PaginationPagesNumberDisconnected extends React.Component<INavigationPaginationProps, any> {
+    private handlePageClick = (pageNb: number) => {
+        if (pageNb >= 0 && this.props.currentPage !== pageNb) {
+            this.props.onPageClick?.(pageNb);
+        }
+    };
+
+    componentDidUpdate() {
+        if (this.props.currentPage > this.props.totalPages - 1) {
+            this.handlePageClick(this.props.totalPages - 1);
+        }
+    }
+
+    componentDidMount() {
+        this.props.onRender?.();
+    }
+
+    componentWillUnmount() {
+        this.props.onDestroy?.();
+    }
+
+    render() {
+        const currentPage: number = this.props.currentPage || 0;
+
+        const showXPages: number = Math.abs((this.props.numberOfPagesToShow || NUMBER_OF_PAGES_SHOWING) - 1);
+        const previousLabel: string = this.props.previousLabel || PREVIOUS_LABEL;
+        const nextLabel: string = this.props.nextLabel || NEXT_LABEL;
+        let start: number = 0;
+        let end: number = showXPages;
```

Comment:
`Math.abs((this.props.numberOfPagesToShow || NUMBER_OF_PAGES_SHOWING) - 1);` equals `showXpages` equals `end`
Is it because you feel it is clearer that way ?

Reply:
showXpages is used below I think. I only copy paste this section from the old pagination without trying to rework because I have already alot of changes :)